Port to python3 and pytest as test runner

pypingou opened the pull request.

pypingou committed:
Port the code to python3

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Make pytest the official test runner

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>
Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp commented:
The Python 3 changes look a lot like `2to3`'s doing... Some of the changes are unnecessary. I'll take a closer look tomorrow. Also, I can't seem to assign this to myself, do I need to be added to the repo?

pypingou commented:
I've used ``2to3`` to get me started on the conversion, I admit not having spent much time reviewing its work afterwards.
I'll give you commit on my fork so you can tweak the PR as you see fit.

nphilipp committed:
More Python3 fixes

- use dict, list, set literals and comprehensions where sensible
- don't build lists from generators superfluously
- don't use dict.keys(), rather the dict var itself where an iterable is
  expected
- sorted() returns a list already
- remove Python 2 compat
- use f-strings, except for logging, don't use % operator there, for
  late interpolation
- remove u-strings from comments, too

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp commented:
I have more fixes in queue, but they're not really related to Python 3, so I've kept them in another branch.

pypingou committed:
Port the Dockerfile used by travis to use python3

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>
